### PR TITLE
[mono][interp] Replace sign extending moves to normal moves

### PR DESCRIPTION
**Pull request by BrzVlad**
If the source var doesn't have its address taken. These opcodes are very frequent (even over 5%) and they cannot be optimized out and block copy propagation.

### CODE REVIEW

File: src/mono/mono/mini/interp/transform.c [hunk 1]
```diff
@@ -8587,6 +8581,13 @@ interp_cprop (TransformData *td)
 				local_defs [dreg].def_index = ins_index;
 			}
 
+			// We always store to the full i4, except as part of STIND opcodes. These opcodes can be
+			// applied to a local var only if that var has LDLOCA applied to it
+			if ((opcode >= MINT_MOV_I1 && opcode <= MINT_MOV_U2) && !td->locals [sregs [0]].indirects) {
```
Comment (vargaz) [inline src/mono/mono/mini/interp/transform.c:8586]: Should it be 'not applied to it' ?

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/interp/transform.c (hunk 1 above)
Comment (BrzVlad) [reply src/mono/mono/mini/interp/transform.c:8586]: I was refering to the STIND opcodes, in the sense that they can store to a local var only if the local var has its address taken.

**Comment by radekdoulik** [comment]
nice! I see improvements in Span measurements around the time this change landed:  https://alinpahontu2912.github.io/testpages/?startDate=2022-09-23T14%3A41%3A03.252Z&endDate=2022-09-30T14%3A41%3A03.252Z&tasks=Span
image

cc @SamMonoRT

**Pull request merged**